rpmlint/checks/BinariesCheck.py

```diff
@@ -140,12 +140,15 @@ def _check_shared_library(self, pkg, pkgfile_path, path):
                 self.output.add_info('E', pkg, 'unused-direct-shlib-dependency',
                                      path, dependency)
 
-    def _check_opt_library_dependency(self, pkg, pkgfile_path, path):
+    def _check_library_dependency_location(self, pkg, pkgfile_path, path):
+        invalid_prefixes = ['/opt/', '/usr/']
```

Comment:
This does not sound correct. It should now error out even if lib in /usr links against a lib in /usr.

I suppose the pr checks for the usr linking if the stuff is in /lib64 and still links against /usr/lib64...